### ISSUE TITLE
Timbre difference between 2.3d-sp3?

### ISSUE THREAD

**Issue by bryc** [8bitbubsy/ahx2play#4]
In this song here: https://modland.com/pub/modules/AHX/Yelm/oldskoolcrackintrosong.ahx

I'm noticing an instrument sounds thinner / more nasal in ahx2play than WinUAE.
It's subtle, but definitely seems like a possible issue.

Instrument is the arpeggio used in Position 8.

Here's ahx2play (unmute both before playing):

https://github.com/8bitbubsy/ahx2play/assets/1408749/c2c1808c-d454-4a92-bed9-27ef700f94a5

Here's AHX 2.3d-sp3 running in WinUAE:

https://github.com/8bitbubsy/ahx2play/assets/1408749/fac8115d-6a7e-4fde-8d08-9d2936a65e37

WinUAE settings that appear closest to ahx2play (A500 filter was duller than ahx2play initially):

![image](https://github.com/8bitbubsy/ahx2play/assets/1408749/867695d1-fc8c-4c67-b50b-3b0d77a26424)

**Comment by 8bitbubsy** [comment]
Thanks for a very thorough bug report!
This is definitely related to the code, I'll have a look today and come back with more information.

**Comment by 8bitbubsy** [comment]
Aha, I was wrong. It's related to the same instrument being played on two channels at once. Paula has a small delay between channels, while my Paula emulator does not. This causes a different phase on real Amiga and WinUAE, while the phase is identical on both channels in ahx2play. If you press play many times in AHX on position 8, you'll even hear the sound changing sometimes.

**Comment by 8bitbubsy** [comment]
Try playing this one over and over, you should hear a phasing issue: https://16-bits.org/etc/phasetest.ahx
At least I do in WinUAE with the 68020 version of AHX 2.3d-sp3.

There's nothing reasonable I can do about this...

**Comment by 8bitbubsy** [comment]
Well, it's different in WinUAE on my end. I use WinUAE 4.10.1, with a stereo separation of 20%.

**Comment by bryc** [comment]
TLDR: A bunch of observations, but I think you already came to the same conclusion. Seems to not be a bug, just a side effect of not having that delay in Paula emulation.

Yeah, I can reproduce that gradual phase effect in 4.10.1, but not 3.0.0. Made a lil animation:

![winUAE ahx phase effect](https://github.com/8bitbubsy/ahx2play/assets/1408749/bf46d723-aac3-4482-b441-1290231ac971)

This is the first few waveform cycles at the start of each pattern iteration over a period of 3 minutes.

It's definitely applying some sort of gradual phase offset similar to this (two square waves with one becoming increasingly out of phase each iteration):
![square phase](https://github.com/8bitbubsy/ahx2play/assets/1408749/a1493c1e-73d1-4fc1-902f-b8b18bb48833)

I can confirm that this also doesn't occur in ahx2play, but **something similar is definitely happening**.

In the actual song, when position 8 comes up, the waveform is definitely phase shifted in a similar way, that doesn't occur in your phasetest.ahx by itself.

A: phasetest.ahx:

![image](https://github.com/8bitbubsy/ahx2play/assets/1408749/e17e635d-9ee0-4ecb-8e14-53197d4fad08)

B: The equivalent, position 8 of original song, but the irrelevant channels silent:

![image](https://github.com/8bitbubsy/ahx2play/assets/1408749/5053ea56-685d-4455-881d-1bdb196c88fa)

Looking closer at the cycle level, a similar stairstep pattern is visible in B (second row):

![image](https://github.com/8bitbubsy/ahx2play/assets/1408749/3a5459e4-363d-4b9f-9cd3-63e32b69316a)

_Side note: I inverted the polarity in this image to match the WinUAE observed behavior - for illustrative purposes. ahx2play appears to render the staircase pattern upside down in this case._

I'm assuming it should sound exactly as it does in phasetest.ahx, because this phasing shouldn't happen?

*****

In any case it does appear that the timbre at that point in the song is similar to ahx2play in WinUAE 4.10.1:

WinUAE 4:

https://github.com/8bitbubsy/ahx2play/assets/1408749/2f2f0b6c-8cd4-4149-aa3c-95693bbbf340

ahx2play:

https://github.com/8bitbubsy/ahx2play/assets/1408749/3b29125b-e8f9-45f8-85b7-8ac50fdc6462

And for completeness, here's phasetest on both. So definitely something alters how position 8 sounds for both, regardless of the Paula delay you mention.

WinUAE 4:

https://github.com/8bitbubsy/ahx2play/assets/1408749/19794497-cb9b-4191-a62b-c234d2975440

ahx2play:

https://github.com/8bitbubsy/ahx2play/assets/1408749/8c8fa091-385d-4abe-91ca-a0625ea8c7e1

So probably not a bug, but without the delay in Paula, it does sound a bit hollow.
